app/views/collection_numbers/_form.html.erb

```diff
@@ -1,9 +1,12 @@
-<% # form_tag(add_query_param(action: :new, id: @observation.id)) do %>
-<%= form_with model: :collection_number do |f| %>
+<% id = @observation&.id
+   id_path_segment = id ? "/#{id}" : nil
+   url = "#{collection_numbers_path}#{id_path_segment}" +
+         "?back=#{@back}&q=#{get_query_param}" %>
 
+<%= form_with model: @collection_number, url: url do |f| %>
   <%= f.submit button.t, class: "btn mx-auto mt-3" %>
 
-  <%= f.hidden_field :id, value: @observation.id %>
+  <%= f.hidden_field :id, value: id %>
   <%= f.hidden_field :q, value: get_query_param %>
```

Comment:
Do we also need a hidden field for `back`?

Reply:
That sounds right. I think i did have that at one point... iirc the controller provides it too.

Reply:
I don't think we need an explicit hidden `back` field; the following already passes:
```ruby
    # Prove that edit passes "back" and query param through to form.
    get(:edit, params.merge(back: "foo", q: q))
    assert_select("form:match('action', ?)", /\?.*q=#{q}/)
    assert_select("form:match('action', ?)", /\?.*back=foo/)
```